cmd/rpcdaemon/commands/engine_api.go

```diff
@@ -246,6 +254,51 @@ func (e *EngineImpl) GetPayloadBodiesV1(ctx context.Context, blockHashes []rpc.B
 	return blockHashToBody, nil
 }
 
+// Gets a transistionConfiguration and pings the execution layer and checks if the execution layer has the correct configurations
+func (e *EngineImpl) ExchangeTransitionConfigurationV1(ctx context.Context, transitionConfiguration TransitionConfiguration, terminalBlockNumber hexutil.Uint64 /* terminalBlockNumber is always zero */) (TransitionConfiguration, error) {
+	tx, err := e.db.BeginRo(ctx)
+
+	if err != nil {
+		return TransitionConfiguration{}, err
+	}
+
+	defer tx.Rollback()
+
+	if terminalBlockNumber != 0 {
+		return TransitionConfiguration{}, fmt.Errorf("received the wrong terminal block number. expected zero, but instead got: %d", terminalBlockNumber)
+	}
+
+	chainConfig, err := e.BaseAPI.chainConfig(tx)
+
+	if err != nil {
+		return TransitionConfiguration{}, err
+	}
+
+	terminalTotalDifficulty := chainConfig.TerminalTotalDifficulty
+
+	if terminalTotalDifficulty != transitionConfiguration.TerminalTotalDifficulty.ToInt() {
+		return TransitionConfiguration{}, fmt.Errorf("the execution layer has the wrong total terminal difficulty. expected %d, but instead got: %d", transitionConfiguration.TerminalTotalDifficulty.ToInt(), terminalTotalDifficulty)
+	}
+
+	if *chainConfig.TerminalBlockHash == (common.Hash{}) {
+		return TransitionConfiguration{
+			TerminalTotalDifficulty: (*hexutil.Big)(terminalTotalDifficulty),
+			TerminalBlockHash:       *chainConfig.TerminalBlockHash,
+			TerminalBlockNumber:     0,
+		}, nil
+	}
+
+	if *chainConfig.TerminalBlockHash != transitionConfiguration.TerminalBlockHash {
```

Comment:
check if `chainConfig.TerminalBlockHash == nil`, that will do so that you do not use erroneously a nil value and panic the daemon